api/src/devcomp/domain/usecases/verify-and-save-answer.js

```diff
@@ -0,0 +1,36 @@
+import { NotFoundError } from '../../../shared/domain/errors.js';
+import { PassageDoesNotExistsError } from '../errors.js';
+
+async function verifyAndSaveAnswer({
+  userResponse,
+  elementId,
+  passageId,
+  passageRepository,
+  moduleRepository,
+  elementAnswerRepository,
+}) {
+  const passage = await _getPassage({ passageId, passageRepository });
+  const module = await moduleRepository.getBySlugForVerification({ slug: passage.moduleId });
+  const element = module.getElementById(elementId);
+  const grain = module.getGrainByElementId(elementId);
```

Comment:
Proposition, d'abord chercher le grain, et chercher l'element dans le grain, cela évite le flatMap qui va reparcourir tous les autres grains alors que vous connaissez déjà le grain cible.

```suggestion
  const grain = module.getGrainByElementId(elementId);
  const element = grain.getElementById(elementId);
 ```
 
 Note : il y a peut-être à créer la fonction `grain.getElementById`